src/saveload/saveload.cpp

```diff
@@ -2982,6 +2982,43 @@ SaveOrLoadResult SaveOrLoad(const std::string &filename, SaveLoadOperation fop,
 	}
 }
 
+/**
+ * Create an autosave or netsave.
+ * @param counter A pointer to the counter variable to be used for rotating the file name
+ * @param netsave indicates if this is a regular autosave or a netsave
+ */
+void DoAutoOrNetsave(int* counter, bool netsave) {
+	char buf[MAX_PATH];
+
+	if (_settings_client.gui.keep_all_autosave) {
+		GenerateDefaultSaveName(buf, lastof(buf));
+		if (!netsave) {
+			strecat(buf, ".sav", lastof(buf));
+		}
+		else {
```

Comment:
[Coding style](https://wiki.openttd.org/en/Development/Coding%20style)
```suggestion
		} else {
```